src/main/java/org/broadinstitute/hellbender/engine/filters/MetricsReadFilter.java

```diff
@@ -0,0 +1,38 @@
+package org.broadinstitute.hellbender.engine.filters;
+
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+public final class MetricsReadFilter implements ReadFilter {
+  // TODO: Should this be something more unique, such as a timestamp, in order
+  // to behave with Spark's serialization?
+  static final long serialVersionUID = 1L;
+
+  private final boolean pfReadOnly;
+  private final boolean alignedReadsOnly;
+
+  /**
+   * @param alignedReadsOnly If set to true calculate mean quality over aligned
+   * reads only.
+   * @param pfReadOnly If set to true calculate mean quality over passing filter
+   * reads only.
+   */
+  public MetricsReadFilter(final boolean pfReadOnly,
```

Comment:
> akiezun added a note 8 days ago
The comment is incorrect if this code is extracted to a common location.
-- https://github.com/broadinstitute/gatk/pull/1053#discussion-diff-43075650

I thought it would still apply with this, but I'm probably missing something about what you meant.